src/panels/lovelace/elements/hui-service-button-element.ts

```diff
@@ -40,12 +40,17 @@ export class HuiServiceButtonElement
       return nothing;
     }
 
+    const updatedTarget = this._config.target ?? {
+      entity_id: this._config.service_data?.entity_id,
```

Comment:
This should not be needed, targets can technically be in the data of a service call too.

(If we would do this, we should do it for all targets, so also area, device, etc)

Reply:
Seems like it's somehow required after all: adding other keys.